### ISSUE TITLE
Stabilise training

### ISSUE THREAD

**Issue by alex-hh** [alex-hh/profam#74]
Some ideas:

disable dropout - what michael suggested?

• Use max learning rate of 6e-5 instead of 1e-4
• Use a gradient clipping value of 0.3 instead of 1.0
• Fully shuffle data
• Use a different seed to ensure different initialization and data order
• Reintroduce LayerNorm at embedding layer (LNem)
• Use a longer learning rate warm-up period of 1800 steps
• Remove incorrect use of weight decay on LayerNorm multipliers (γ
•
• )
• Use Megatron initialization rescaling (see use of z
′
in Appendix A.5)
• Apply query key layer scaling (Shoeybi et al., 2019)
• Apply a batch size warm-up: Use a batch size of 1024 for 7200 iterations, then increase
to 2048

https://arxiv.org/pdf/2303.17564
